packages/webapp/src/containers/Finances/saga.js

```diff
@@ -148,6 +151,69 @@ export function* getFarmExpenseTypeSaga() {
   }
 }
 
+export function* addCustomExpenseTypeSaga(action) {
+  const { expenseTypeUrl } = apiConfig;
+  let { user_id, farm_id } = yield select(loginSelector);
+  const header = getHeader(user_id, farm_id);
+  const { expense_name } = action.custom_expense_type;
+
+  try {
+    const result = yield call(axios.post, expenseTypeUrl, { farm_id, expense_name }, header);
+    if (result) {
+      yield put(enqueueSuccessSnackbar(i18n.t('message:EXPENSE_TYPE.SUCCESS.ADD')));
+      yield call(getFarmExpenseTypeSaga);
+      history.push('/manage_custom_expenses');
+    }
+  } catch (e) {
+    console.log('failed to add new expense type to the database');
+    yield put(enqueueErrorSnackbar(i18n.t('message:EXPENSE_TYPE.ERROR.ADD')));
+  }
+}
+
+export function* updateCustomExpenseTypeSaga(action) {
+  const { expenseTypeUrl } = apiConfig;
+  let { user_id, farm_id } = yield select(loginSelector);
+  const header = getHeader(user_id, farm_id);
+  const { expense_type_id } = action;
+  const { expense_type_name } = action.custom_expense_type;
```

Comment:
And this and the API call right below need to both be `expense_name`, not `expense_type_name`. Then the route will work!

Reply:
Thank you!